### PR TITLE
DOCSUP-5822: Document the accurateCastOrNull function.

### PR DESCRIPTION
**Pull request by ka1bi4**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Documentation (changelog entry is not required)

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

- Documentation for #16724

Detailed description / Documentation draft:

...

### CODE REVIEW

File: docs/en/sql-reference/functions/type-conversion-functions.md
Comment (olgarev) [inline docs/en/sql-reference/functions/type-conversion-functions.md:496]:
```suggestion
Converts `x` to the `T` data type. The difference from [cast(x, T)](#type_conversion_function-cast) is that `accurateCast`
does not allow overflow of numeric types during cast if type value `x` does not fit bounds of type `T`. If the casted value is not representable in the target type, `accurateCast` throws an exception.
```
```suggestion
Converts `x` to the `T` data type. The differente from [cast(x, T)](#type_conversion_function-cast) is that `accurateCast`
does not allow overflow of numeric types during cast if type value `x` does not fit bounds of type `T`.
```

File: docs/en/sql-reference/functions/type-conversion-functions.md
Comment (olgarev) [inline docs/en/sql-reference/functions/type-conversion-functions.md:431]: Note, that if value `x` does not fit the bounds of type `T`, the function will overflow. For example, `CAST(-1, 'UInt8')` will return `255`.

File: docs/en/sql-reference/functions/type-conversion-functions.md
Comment (olgarev) [inline docs/en/sql-reference/functions/type-conversion-functions.md:564]:
```suggestion
``` sql
SELECT
    accurateCastOrNull(-1, 'UInt8') as uint8,
    accurateCastOrNull(128, 'Int8') as int8,
    accurateCastOrNull('Test', 'FixedString(2)') as fixed_string;
```
```

File: docs/en/sql-reference/functions/type-conversion-functions.md
Comment (olgarev) [inline docs/en/sql-reference/functions/type-conversion-functions.md:563]:
```suggestion
    accurateCastOrNull(-1, 'UInt8') as uint8,
    accurateCastOrNull(128, 'Int8') as int8,
    accurateCastOrNull('Test', 'FixedString(2)') as fixed_string;
```

File: docs/ru/sql-reference/functions/type-conversion-functions.md
Comment (olgarev) [inline docs/ru/sql-reference/functions/type-conversion-functions.md:500]:
```suggestion
**Параметры**
```

File: docs/en/sql-reference/operators/in.md
Comment (olgarev) [inline docs/en/sql-reference/operators/in.md:20]:
```suggestion
ClickHouse allows different types in the left and right parts of `IN` subquery. In this case it converts the left hand side to the type of the right hand side as if the [accurateCastOrNull](../functions/type-conversion-functions.md#type_conversion_function-accurate-cast_or_null) function is applied.
```

File: docs/en/sql-reference/functions/type-conversion-functions.md
Comment (olgarev) [inline docs/en/sql-reference/functions/type-conversion-functions.md:549]:
```suggestion
-   The value, converted to the specified data type `T`.
```

File: docs/ru/sql-reference/functions/type-conversion-functions.md
Comment (olgarev) [inline docs/ru/sql-reference/functions/type-conversion-functions.md:426]:
```suggestion
Преобразует входное значение `x` в указанный тип данных `T`.
```

File: docs/ru/sql-reference/functions/type-conversion-functions.md
Comment (olgarev) [inline docs/ru/sql-reference/functions/type-conversion-functions.md:430]:
```suggestion
Обратите внимание, что если значение `x` не может быть преобразовано к типу `T`, возникает переполнение. Например, `CAST(-1, 'UInt8')` возвращает 255.
```

File: docs/ru/sql-reference/functions/type-conversion-functions.md
Comment (olgarev) [inline docs/ru/sql-reference/functions/type-conversion-functions.md:497]:
```suggestion
В отличие от функции [cast(x, T)](#type_conversion_function-cast), `accurateCast` не допускает переполнения при преобразовании числовых типов. Например, `accurateCast(-1, 'UInt8')` вызовет исключение.
```

File: docs/ru/sql-reference/functions/type-conversion-functions.md
Comment (olgarev) [inline docs/ru/sql-reference/functions/type-conversion-functions.md:530]:
```suggestion
Всегда возвращает тип [Nullable](../../sql-reference/data-types/nullable.md). Если исходное значение не может быть преобразовано к целевому типу, возвращает [NULL](../../sql-reference/syntax.md#null-literal).
```

File: docs/ru/sql-reference/operators/in.md
Comment (olgarev) [inline docs/ru/sql-reference/operators/in.md:21]:
```suggestion
Если типы данных в левой и правой частях подзапроса `IN` различаются, ClickHouse преобразует значение в левой части к типу данных из правой части. Преобразование выполняется по аналогии с функцией [accurateCastOrNull](../functions/type-conversion-functions.md#type_conversion_function-accurate-cast_or_null), т.е. тип данных становится [Nullable](../../sql-reference/data-types/nullable.md), а если преобразование не может быть выполнено, возвращается значение [NULL](../../sql-reference/syntax.md#null-literal).
```

File: docs/en/sql-reference/operators/in.md
Comment (olgarev) [inline docs/en/sql-reference/operators/in.md:20]:
```suggestion
ClickHouse allows types to differ in the left and the right parts of `IN` subquery. In this case it converts the left side value to the type of the right side, as if the [accurateCastOrNull](../functions/type-conversion-functions.md#type_conversion_function-accurate-cast_or_null) function is applied. That means, that the data type becomes [Nullable](../../sql-reference/data-types/nullable.md), and if the conversion cannot be performed, it returns  [NULL](../../sql-reference/syntax.md#null-literal).
```

**Comment by ka1bi4** [comment]
Hi,
PR is ready for review.
I updated documentation for accurateCastOrNull function and IN operator.
Moreover, fix some query examples. 

After merging master, I got some changes in reinterpretAs(x, T) — it's not my modifications. Maybe, I should marged it again?

File: docs/ru/sql-reference/functions/type-conversion-functions.md
Comment (KochetovNicolai) [inline docs/ru/sql-reference/functions/type-conversion-functions.md:512]: ┘

**Review by KochetovNicolai — APPROVED** [review]

File: docs/en/sql-reference/functions/type-conversion-functions.md
Comment (KochetovNicolai) [inline docs/en/sql-reference/functions/type-conversion-functions.md:306]: This variant still works for me. It is strange it was removed.

**Comment by olgarev** [comment]
@alexey-milovidov Hi, this PR was approved but not merged into master. Can we close the ticket?

**Comment by KochetovNicolai** [comment]
Well, there is another one pr about reinterpret #20611
Probably, we should merge it first, and then merge docs changes here.

**Comment by olgarev** [comment]
> Well, there is another one pr about reinterpret #20611
> Probably, we should merge it first, and then merge docs changes here.

@KochetovNicolai Hi! The PR we were waiting for is merged now.

**Comment by KochetovNicolai** [comment]
@olgarev Hi!
Let's resolve conflicts in `type-conversion-functions.md`, and then it may be merged.
Do you need some help with it?

**Comment by ka1bi4** [comment]
@KochetovNicolai Hi!
I resolved the conflict.

**Pull request merged**
